Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Performance.hs

```diff
@@ -19,15 +24,37 @@ data Results = Results
   }
   deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
 
-newtype PerformanceRes = PerformanceRes
-  { referrals :: Results
+data PerformanceRes = PerformanceRes
+  { referrals :: Results,
+    referralCode :: Text,
+    totalDriver :: Int,
+    currRideRank :: Integer,
+    currReferralRank :: Integer
   }
   deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
 
 getDriverPerformance :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => (Id SP.Person, Id DM.Merchant) -> m PerformanceRes
-getDriverPerformance (driverId, _) = do
+getDriverPerformance (driverId, merchantId) = do
   _ <- B.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
   -- _ <- QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
   allRefferedCustomers <- QRD.findAllReferredByDriverId driverId
   let ridesTakenList = filter (.hasTakenValidRide) allRefferedCustomers
-  pure $ PerformanceRes (Results (length allRefferedCustomers) (length ridesTakenList))
+  driverReferralCode <- fmap (.referralCode) <$> QDR.findById (cast driverId)
+  nowUtc <- getCurrentTime
+  let today = RideEndInt.getCurrentDate nowUtc
+
+  mbCurrPersonRideRank <- Redis.withNonCriticalRedis $ Redis.zRevRank (RideEndInt.makeDailyDriverLeaderBoardKey merchantId today LConfig.RIDE) driverId.getId
+  currPersonRideRank <- case mbCurrPersonRideRank of
+    Nothing -> Redis.withNonCriticalRedis $ Redis.zCard (RideEndInt.makeDailyDriverLeaderBoardKey merchantId today LConfig.RIDE)
+    Just rank -> pure rank
+
+  mbCurrPersonReferralRank <- Redis.withNonCriticalRedis $ Redis.zRevRank (RideEndInt.makeDailyDriverLeaderBoardKey merchantId today LConfig.REFERRAL) driverId.getId
+  currPersonReferralRank <- case mbCurrPersonReferralRank of
+    Nothing -> Redis.withNonCriticalRedis $ Redis.zCard (RideEndInt.makeDailyDriverLeaderBoardKey merchantId today LConfig.REFERRAL)
+    Just rank -> pure rank
+
+  (active, inactive) <- QDI.totalDrivers merchantId
+
+  case driverReferralCode of
+    Just code -> pure $ PerformanceRes (Results (length allRefferedCustomers) (length ridesTakenList)) code.getId (active + inactive) currPersonRideRank currPersonReferralRank
+    Nothing -> pure $ PerformanceRes (Results (length allRefferedCustomers) (length ridesTakenList)) "" (active + inactive) currPersonRideRank currPersonReferralRank
```

Comment:
`(length allRefferedCustomers)` - can we send 0?

Reply:
no... frontend uses this information that's why we need to send actual value rather than 0

![Screenshot 2023-09-16 at 10 03 47 PM](https://github.com/nammayatri/nammayatri/assets/83864083/e472dea6-3702-431b-a065-a01a85e0552f)

Reply:
I'm talking about the case when referral code is nothing

Reply:
okay got the point. Will send 0 in the case for no referral code.